hail/src/main/scala/is/hail/expr/ir/functions/ArrayFunctions.scala

```diff
@@ -328,8 +329,8 @@ object ArrayFunctions extends RegistryFunctions {
             setup1,
             setup2),
           m1 || m2 || Code(
-            a1 := v1.tcode[Long],
-            a2 := v2.tcode[Long],
+            a1 := v1.asInstanceOf[SIndexablePointerCode].a,
```

Comment:
This isn't correct -- this won't always be an SIndexablePointerCode. The right thing is to use the indexable SType interface here

Reply:
I'll fix this later then. It's a larger change involving more Code[_] -> CodeBuilder.